docs/how-to-guides/setup-thirdparty-auth-for-grafana.md

```diff
@@ -0,0 +1,144 @@
+  component.
+
+## Steps
+
+> **NOTE**: This guide assumes that the OAuth provider is GitHub. For other OAuth providers, the
+> steps are the same, but the secret environment variables will change, as mentioned in [Step
+> 2](#step-2-add-prometheus-operator-component-configuration). Grafana docs explain how to convert
+> the `ini` config to environment variables
+> [here](https://grafana.com/docs/grafana/latest/administration/configuration/#configure-with-environment-variables).
+
+### Step 1: Create Github application
+
+- Create a GitHub OAuth application as documented in the [Grafana
+  docs](https://grafana.com/docs/grafana/latest/auth/github/).
+
+- Set **Homepage URL** to `https://grafana.<cluster name>.<DNS zone>`. This should be same as the
+  `prometheus-operator.grafana.ingress.host` as shown in [Step
+  2](#step-2-add-prometheus-operator-component-configuration).
+
+- Set **Authorization callback URL** to `https://grafana.<cluster name>.<DNS zone>/login/github`.
+
+- Make a note of `Client ID` and `Client Secret`, they will be needed in [Step
+  3](#step-3-add-secret-information).
+
+### Step 2: Add `prometheus-operator` component configuration
+
+Create a file named `prometheus-operator.lokocfg` with the following contents or if you already
+have `prometheus-operator` installed then add the following contents to the existing configuration:
+
+```tf
+variable "gf_auth_github_client_id" {}
+variable "gf_auth_github_client_secret" {}
+variable "gf_auth_github_allowed_orgs" {}
+
+component "prometheus-operator" {
+  grafana {
+    secret_env = {
+      "GF_AUTH_GITHUB_ENABLED"               = "'true'"
+      "GF_AUTH_GITHUB_ALLOW_SIGN_UP"         = "'true'"
+      "GF_AUTH_GITHUB_SCOPES"                = "user:email,read:org"
+      "GF_AUTH_GITHUB_AUTH_URL"              = "https://github.com/login/oauth/authorize"
+      "GF_AUTH_GITHUB_TOKEN_URL"             = "https://github.com/login/oauth/access_token"
+      "GF_AUTH_GITHUB_API_URL"               = "https://api.github.com/user"
+      "GF_AUTH_GITHUB_CLIENT_ID"             = var.gf_auth_github_client_id
+      "GF_AUTH_GITHUB_CLIENT_SECRET"         = var.gf_auth_github_client_secret
+      "GF_AUTH_GITHUB_ALLOWED_ORGANIZATIONS" = var.gf_auth_github_allowed_orgs
+    }
+
+    ingress {
+      host = "grafana.<cluster name>.<DNS zone>"
+    }
+  }
+}
+```
+
+> **NOTE**: On Packet, you either need to create a DNS entry for `grafana.<cluster name>.<DNS zone>`
+> and point it to the Packet external IP for the contour service (see the [Packet ingress guide for
+> more details](./ingress-with-contour-metallb.md)) or use the [External DNS
+> component](../configuration-reference/components/external-dns.md).
+
+> **NOTE**: In the above configuration, boolean values are set to `"'true'"` instead of bare
+> `"true"` because Kubernetes expects the key-value pair to be of type `map[string]string` and not
+> `map[string]bool`.
```

Comment:
Mentioning `map[string]string` and such seems too detailed for a howto, but I don't have a better suggestion...

Reply:
Yeah I thought how best to put it but I think as programmers this makes most sense to convey why it does not work.